sycl/source/detail/online_compiler/online_compiler.cpp

```diff
@@ -160,16 +160,13 @@ compileToSPIRV(const std::string &Source, sycl::info::device_type DeviceType,
                       &SourceName, 0, nullptr, nullptr, nullptr, &NumOutputs,
                       &Outputs, &OutputLengths, &OutputNames);
 
-  byte *SpirV = nullptr;
+  std::vector<byte> SpirV;
   std::string CompileLog;
-  size_t SpirVSize = 0;
   for (uint32_t I = 0; I < NumOutputs; I++) {
     size_t NameLen = strlen(OutputNames[I]);
     if (NameLen >= 4 && strstr(OutputNames[I], ".spv") != nullptr &&
         Outputs[I] != nullptr) {
-      SpirVSize = OutputLengths[I];
-      SpirV = new byte[SpirVSize];
-      std::memcpy(SpirV, Outputs[I], SpirVSize);
```

Comment:
@v-klochkov - Can there only be at most one .spv in `OutputNames`?

Reply:
@v-klochkov, ping, can you answer the question above?

Reply:
The changes look good to me.
Regarding the number of output spirv files. It is assumed to be 1 at most, but I did not find any strict wordings about that in ocloc API https://github.com/intel/compute-runtime/blob/master/shared/offline_compiler/source/ocloc_api.h

It may be a good idea to add an assert here checking that there is only 1 spv output.

Reply:
Good idea! I have added such an assertion.